core/src/main/java/tech/pegasys/eth2signer/core/multikey/DirectoryBackedArtifactSignerProvider.java

```diff
@@ -57,117 +54,73 @@ public DirectoryBackedArtifactSignerProvider(
     this.configsDirectory = rootDirectory;
     this.fileExtension = fileExtension;
     this.signerParser = signerParser;
-    this.artifactSignerCache =
-        CacheBuilder.newBuilder()
-            .maximumSize(maxSize)
-            .build(CacheLoader.from((i) -> loadSignerForIdentifier(i).orElseThrow()));
   }
 
   @Override
   public Optional<ArtifactSigner> getSigner(final String signerIdentifier) {
     final String normalisedIdentifier = normaliseIdentifier(signerIdentifier);
-    final ArtifactSigner signer;
-    try {
-      signer = artifactSignerCache.get(normalisedIdentifier);
-    } catch (UncheckedExecutionException e) {
-      if (e.getCause() instanceof NoSuchElementException) {
-        LOG.error("No valid matching metadata file found for the identifier {}", signerIdentifier);
-      } else {
-        LOG.error("Error loading for signer for identifier {}", signerIdentifier);
-      }
-      return Optional.empty();
-    } catch (Exception e) {
-      LOG.error("Error loading for signer for identifier {}", signerIdentifier);
-      return Optional.empty();
-    }
+    final Optional<ArtifactSigner> result =
+        Optional.ofNullable(artifactSigners.get(normalisedIdentifier));
 
-    if (!signerMatchesIdentifier(signer, signerIdentifier)) {
-      LOG.error(
-          "Signing metadata config does not correspond to the specified signer identifier {}",
-          signer.getIdentifier());
-      return Optional.empty();
+    if (result.isEmpty()) {
+      LOG.error("No signer was loaded matching identifitier '{}'", signerIdentifier);
     }
-    return Optional.of(signer);
+    return result;
   }
 
   @Override
   public Set<String> availableIdentifiers() {
-    return allIdentifiers()
+    return artifactSigners
+        .keySet()
         .parallelStream()
-        .map(this::getSigner)
-        .flatMap(Optional::stream)
-        .map(ArtifactSigner::getIdentifier)
-        .collect(Collectors.toSet());
-  }
-
-  private Set<String> allIdentifiers() {
-    final Function<Path, String> getSignerIdentifier =
-        file -> FilenameUtils.getBaseName(file.toString());
-    return findSigners(this::matchesFileExtension, getSignerIdentifier).stream()
-        .filter(Objects::nonNull)
-        .map(identifier -> "0x" + normaliseIdentifier(identifier))
+        .map(id -> "0x" + id)
```

Comment:
nit: `!id.startsWith("0x") || !is.startsWith("0X") ? "ox" + id : id`. This will handle cases where id already have ox prefix. (or better use hasPrefix kind of method instead).

Reply:
going to ignore 0X as I'd argue that's not a valid representation of a hex prefix.
Also - we _probably_ shouldn't ever mod the ID reported by the signer ... it should be passed through ... but don't want to deal with that now.

Reply:
Given it is known that the artifacSigners are indexed by normalisedIdentifier - which definitively does not have an 0x, is there any need to do this check?

Reply:
If that's the assumption being made, is loadSigners is now mandatory? Should loadSigners be called in the constructor as without this it's not a valid ArtifactSignerProvider?